include/mxnet/ndarray.h

```diff
@@ -284,14 +317,23 @@ class NDArray {
     if (is_none()) return kUndefinedStorage;
     return ptr_->storage_type;
   }
-  inline size_t num_aux() const {
-    if (is_none()) return 0;
-    return ptr_->aux_handles.size();
-  }
   /*! \return whether this ndarray is not initialized */
   inline bool is_none() const {
     return ptr_.get() == nullptr;
   }
+  // returns true if a sparse ndarray's aux_data are not initialized,
+  // indicating a NDArray with zeros
+  inline bool is_zeros_hint() const {
+    if (is_none()) return true;
+    auto stype = storage_type();
+    CHECK_NE(stype, kDefaultStorage);
+    if (stype == kRowSparseStorage || stype == kCSRStorage) {
+      return aux_shape(0).Size() == 0;
```

Comment:
Yes that's why I commented that it's just a hint for whether the ndarray is initialized thus containing zeros. I assume this function will be called quite often implementing various sparse operators. indptr[num_rows] is more accurate but that involves reading memory from GPU, which is more expensive. 
Usually csr is some user input which is non-zero. So this check works for most of the cases. Maybe I should change the change to `storage_is_initialized` to avoid confusion?